src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -635,14 +635,10 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                 // name object
                 $objtype = $char;
                 ++$offset;
-                $pregResult = preg_match(
-                    '/^([^\x00\x09\x0a\x0c\x0d\x20\s\x28\x29\x3c\x3e\x5b\x5d\x7b\x7d\x2f\x25]+)/',
-                    substr($pdfData, $offset, 256),
-                    $matches
-                );
-                if (1 == $pregResult) {
-                    $objval = $matches[1]; // unescaped value
-                    $offset += \strlen($objval);
+                $span = strcspn($pdfData, "\x00\x09\x0a\x0c\x0d\x20\n\t\r\v\f\x28\x29\x3c\x3e\x5b\x5d\x7b\x7d\x2f\x25", $offset, 256);
+                if ($span > 0) {
```

Comment:
That's first in the code, but it was the last case i discovered.
We get not a tiny substring (and allocate memory) just to feed it to regexp.
And we use regexp just to get longest string, withous prohibited ones.

That's what strcspn with offset param do! (and max len, if you really need it)

Note, that 0x20\s fragment in regexp has transformed to a list of prohibited chars: \x20\n\t\r\v\f